### DISCUSSION TITLE
Sharable Config Presets -> Self-hosted GitHub Question

### DISCUSSION THREAD

**Discussion by JoeJennings** [renovatebot/renovate#15409] [Request Help] [answered]
**Question:**
Is there a way to configure Renovate so that a shareable config repository `renovate-config` can have a GitHub visibility of `Internal` and repositories in a different GitHub organization than the shareable config repository can be onboarded using the shareable config? 

**Context:**
I have a GitHub organization say orgA that contains the sharable config in a `renovate-config` repository containing the `default.json`.  The GitHub visibility of the `renovate-config` repo is set to `Internal`.  I have another GitHub organization say orgB that contains a repository I'm attempting to onboard into Renovate.  Renovate is unable to access the shareable config from the orgA repository when attempting to onboard the repository in orgB.

Items completed in hopes of resolving issue:
- Renovate is onboarded into both GitHub organizations orgA & orgB
- The orgA shareable config repository is onboarded to Renovate 
- The GitHub user Renovate assumes identify of is added to both orgs as an Admin
- The personal access token for the Renovate GitHub user is given `notification, read:org, repo, user, and workflow` scopes/access

**Observations:**
Renovate is able to access the sharable config when onboarding repositories in orgA, but not when onboarding repositories in other GitHub orgs.
If the sharable config repository visibility is changed to `Public`, Renovate is able to access the config when onboarding repositories in other GitHub orgs.  However this is not the desired visibility for this repository. 

If the answer to the question is: **Yes**, guidance on how to go about this would be a help.  **Thanks in advance for your time.**

**Comment by rarkins** [comment, marked as answer]
It sounds like GitHub is denying access to the other repo, rather than Renovate deciding not to access it. Which makes the answer "No, unless you can figure out why GitHub is denying access"

**Reply by JoeJennings** [reply]
Yes, in the Renovate logs I can see Renovate is attempting to access the configuration repository/file I believe using the GitHub API and is getting back a 404. I suspect this has to do with Renovate perhaps not sending an authentication token when calling the API or GitHub not liking the token being sent or something along those lines.  I do find it curious that the configuration repository/file can be accessed by Renovate when onboarding repositories within the same GitHub organization that the configuration repository/file resides, this leads me to believe that Renovate is accessing the configuration in a different manner in that scenario.

**Reply by viceice** [reply]
how are you running renovate? the hosted app uses organization scoped app token, so no access to other organizations private repos.

btw: what's the reason to make the preset private?

**Reply by JoeJennings** [reply]
Running self-hosted app, GitHub `Internal` visibility must be same/similar as `Private` as far as token authentication goes, which makes sense I suppose.  The reasons for `Internal` visibility I as understand them are 2 parts, the 1st is more general not specific to any GitHub Enterprise organization/repository,  the `Internal` visibility is being used to apply controls on access. Third party consultants are leveraged, but are not given full access to GitHub organizations/repositories.  The 2nd part is for consistency across a multitude of organizations/repositories with regard to their configuration/settings.  

@viceice  and @rarkins  -> Thank you both for your time and insights.

**Comment by straub** [comment]
In case it's helpful to future readers: I was able to work around this issue by making sure the self-hosted GitHub app was also installed to the org containing the shared preset, generating a second token with that installation, and configuring it separately in `hostRules`, e.g.
```js
hostRules.push({
  hostType: 'github',
  matchHost: `https://api.github.com/repos/${orgName}/`,
  token,
});
```
This way, when Renovate calls the API to fetch the preset from `${orgName}`, it's authenticated and can succeed instead of hitting that 404 error.
